amity/Tests/test.py

```diff
@@ -0,0 +1,239 @@
+        self.amity.add_person('FELLOW', 'test_fellow', 'N')
+        message = sys.stdout.getvalue().strip()
+        self.assertIn('test_fellow already exists', message)
+
+    def test_fellow_is_allocated_a_living_space(self):
+        """Tests that a fellow member is allocated a living space"""
+        self.amity.create_room('LIVINGSPACE', 'test_livingspace')
+        self.amity.add_person('FELLOW', 'Ivy', 'Y')
+        message = sys.stdout.getvalue().strip()
+        self.assertIn('Ivy successfully added to living space test_livingspace',
+                      message)
+
+    def test_create_office(self):
+        """Tests that a new office is created successfully"""
+        initial_office_count = len(self.amity.all_rooms['office'])
+        # initial length of office list before creating a new office
+        self.amity.create_room('OFFICE', 'newoffice')
+        new_office_count = len(self.amity.all_rooms['office'])
+        # new length of office list after creation of office
+        self.assertGreater(new_office_count, initial_office_count)
+
+    def test_cannot_create_office_with_same_name(self):
+        """Tests that a existing office cannot be created"""
+        self.amity.create_room('OFFICE', 'test_office')
+        self.amity.create_room('OFFICE', 'test_office')
+        message = sys.stdout.getvalue().strip()
+        self.assertIn('test_office already exists', message)
+
+    def test_cannot_create_office_with_wrong_room_type(self):
+        """Tests that a new office is not created when wrong room type is
+           inputted"""
+        self.amity.create_room('Wrong_input', 'test_wrong_office')
+        message = sys.stdout.getvalue().strip()
+        self.assertIn('Invalid room type', message)
+
+    def test_create_livingspace(self):
+        """Tests that a new living space is created successfully"""
+        initial_livingspace_count = len(self.amity.all_rooms['livingspace'])
+        self.amity.create_room('LIVINGSPACE', 'newlivingspace')
+        new_livingspace_count = len(self.amity.all_rooms['livingspace'])
+        self.assertGreater(new_livingspace_count, initial_livingspace_count)
+
+    def test_cannot_create_livingspace_with_same_name(self):
+        """Tests that a existing livingspace cannot be created"""
+        self.amity.create_room('LIVINGSPACE', 'test_livingspace')
+        self.amity.create_room('LIVINGSPACE', 'test_livingspace')
+        message = sys.stdout.getvalue().strip()
+        self.assertIn('test_livingspace already exists', message)
+
+    def test_cannot_create_livingspace_with_wrong_room_type(self):
+        """Tests that a new livingspace is not created when wrong room type is
+           inputted"""
+        self.amity.create_room('Wrong_input', 'test_wrong_livingspace')
+        message = sys.stdout.getvalue().strip()
+        self.assertIn('Invalid room type', message)
+
+    def test_reallocation_of_fellow_to_new_office(self):
```

Comment:
When testing if a fellow has been removed from their previous room, it will be best if you check if they have been removed from the list that holds occupants of  a certain room rather that asserting a message.  i.e 
```
def test_reallocated_staff_is_removed_from_previous_office(self):
 +        """Tests that a staff is removed from the previous office once they are
 +           reallocated to a new office"""
 +        self.amity.create_room('OFFICE', 'Blue')
 +        self.amity.add_person('STAFF', 'Hamzi Wato')
 +        self.amity.create_room('OFFICE', 'Red')
 +        self.amity.reallocate_person('Hamzi Wato', 'Red')
 +        message = sys.stdout.getvalue().strip()
 +        self.assertIn('Hamzi Wato removed from previous office', message)
```